src/main/java/frc/robot/RobotContainer.java

```diff
@@ -278,12 +272,41 @@ private void configureBindings() {
   }
 
   public void periodic() {
+    double startTime = HALUtil.getFPGATime();
+
     DogLog.log("nearest", EagleUtil.closestReefSetPoint(drivetrain.getPose(), 0));
+    // 1
+    DogLog.log(
```

Comment:
Suggestion - collect the time after each step, but don't do the logging of duration until the end of the periodic function.
```
double t0= HALUtil.getFPGATime();
// item 1
double t1 = HALUtil.getFPGATime();
// item 2
double t2 = HALUtil.getFPGATime();
...
DogLog.log("thing-1", (t1 - t0)/1000);
DogLog.log("thing-2", (t2 - t1)/1000);
```